### PR TITLE
[CELEBORN-778] Rename MemoryManagerStat to ServingState

### PR DESCRIPTION
**Pull request by pan3793**
### What changes were proposed in this pull request?

Rename

```
enum MemoryManagerStat { resumeAll, pausePushDataAndReplicate, pausePushDataAndResumeReplicate }
```

to

```
enum ServingState { NONE_PAUSED, PUSH_AND_REPLICATE_PAUSED, PUSH_PAUSED }
```

### Why are the changes needed?

`MemoryManagerStat` indicates the worker serving functionalities, and it's weird to represent the state using verbs.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Pass GA.

### CODE REVIEW

**Comment by pan3793** [comment]
cc @cfmcgrady

File: worker/src/main/java/org/apache/celeborn/service/deploy/worker/memory/MemoryManager.java [hunk 1]
```diff
@@ -264,30 +244,29 @@ private MemoryManager(CelebornConf conf) {
...
-        return MemoryManagerStat.resumeAll;
-      } else {
-        if (underPressure) {
-          return MemoryManagerStat.pausePushDataAndResumeReplicate;
-        } else {
-          return MemoryManagerStat.resumeAll;
-        }
-      }
+    } else if (resume) {
+      underPressure = false;
```
Comment (cfmcgrady) [inline worker/src/main/java/org/apache/celeborn/service/deploy/worker/memory/MemoryManager.java:255]: `underPressure` is only used in this function.

Comment (cfmcgrady) [reply worker/src/main/java/org/apache/celeborn/service/deploy/worker/memory/MemoryManager.java:255]: it seems that we can further simplify this method

**Review by cfmcgrady — APPROVED** [review]
LGTM

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks! cc @FMX , could you please take a look at this, especially ```currentServingState```?

**Comment by pan3793** [comment]
Merged to main/0.3

**Pull request closed without merging**